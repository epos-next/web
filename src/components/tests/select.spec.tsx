import Select from "@components/select";
import { fireEvent, render } from "@testing-library/react";
import React from "react";

it("should match snapshot", () => {
    const { asFragment } = render(<Select/>)
    expect(asFragment()).toMatchSnapshot()
});

it("should display error state", () => {
    const { asFragment } = render(<Select error={ true }/>)
    expect(asFragment()).toMatchSnapshot()
});

it("should select values", () => {
    const callback = jest.fn()
    const { getByRole } = render(
        <Select onChange={ callback }>
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
        </Select>
    )

    fireEvent.change(getByRole("combobox"), { target: { value: "2" } })
    expect(getByRole("combobox")).toHaveValue("2")
})

it("should have default selected value", () => {
    const callback = jest.fn()
    const { getByRole } = render(
        <Select onChange={ callback }>
            <option value="1">1</option>
            <option value="2" defaultChecked disabled hidden>2</option>
            <option value="3">3</option>
        </Select>
    )

    expect(getByRole("combobox")).toHaveValue("1")
})

describe("testing native select props", () => {
    it("should pass placeholder", () => {
        const placeholder = "heey what's up brrrr"
        const { getByRole } = render(<Select placeholder={ placeholder }/>)
        expect(getByRole("combobox")).toHaveAttribute("placeholder", placeholder)
    });

    it("should pass custom attribute", () => {
        const { getByRole } = render(<Select data-attribute="123"/>)
        expect(getByRole("combobox")).toHaveAttribute("data-attribute", "123")
    });
})


