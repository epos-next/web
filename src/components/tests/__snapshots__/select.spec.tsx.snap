// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display error state 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c1::-moz-placeholder {
  color: var(--light-primary);
}

.c1:-ms-input-placeholder {
  color: var(--light-primary);
}

.c1::placeholder {
  color: var(--light-primary);
}

.c1:focus {
  border: 2px solid var(--contrast);
}

.c1[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c1[data-error=true]:focus {
  border: 2px solid var(--error);
}

.c2 {
  position: absolute;
  top: 16px;
  right: 14px;
}

.c0 {
  position: relative;
  width: 100%;
}

.c0 select:invalid {
  color: var(--light-primary);
}

@media screen and (max-width:768px) {
  .c1 {
    padding: 13px;
  }
}

<div
    class="c0"
  >
    <select
      class="c1"
      data-error="true"
    />
    <img
      class="c2"
      src="/icons/arrow-icon.png"
    />
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c1::-moz-placeholder {
  color: var(--light-primary);
}

.c1:-ms-input-placeholder {
  color: var(--light-primary);
}

.c1::placeholder {
  color: var(--light-primary);
}

.c1:focus {
  border: 2px solid var(--contrast);
}

.c1[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c1[data-error=true]:focus {
  border: 2px solid var(--error);
}

.c2 {
  position: absolute;
  top: 16px;
  right: 14px;
}

.c0 {
  position: relative;
  width: 100%;
}

.c0 select:invalid {
  color: var(--light-primary);
}

@media screen and (max-width:768px) {
  .c1 {
    padding: 13px;
  }
}

<div
    class="c0"
  >
    <select
      class="c1"
    />
    <img
      class="c2"
      src="/icons/arrow-icon.png"
    />
  </div>
</DocumentFragment>
`;
